include/mbedtls/check_config.h

```diff
@@ -755,9 +755,8 @@
 #error "MBEDTLS_PSA_CRYPTO_C defined, but not all prerequisites (missing RNG)"
 #endif
 
-#if defined(MBEDTLS_PSA_CRYPTO_C) && \
-    (defined(PSA_HAVE_SOFT_BLOCK_CIPHER) || defined(PSA_HAVE_SOFT_BLOCK_AEAD)) && \
-    !defined(MBEDTLS_CIPHER_C)
+#if defined(MBEDTLS_PSA_CRYPTO_C) && defined(PSA_HAVE_SOFT_BLOCK_MODE) && \
+    defined(PSA_HAVE_SOFT_BLOCK_CIPHER) && !defined(MBEDTLS_CIPHER_C)
```

Comment:
I'm not saying this is not correct, but it's not obvious to me why `PSA_HAVE_SOFT_BLOCK_MODE` was added compared to what we had before (especially with `&&`). Can you explain?

Reply:
> Then the issue I found in leaving `PSA_HAVE_SOFT_BLOCK_CIPHER` is that this symbol is true as soon as at least 1 between AES/ARIA/Camellia is enabled (`PSA_WANT_KEY_TYPE_xxx`) but not accelerated. Then I thought that having the key-type without any supported mode (SOFT mode to be precise) didn't make much sense, so that's why I added `PSA_HAVE_SOFT_BLOCK_MODE` here.

Ah, that's the part I was missing. Thanks for the detailed explanation!